editoast/src/views/path/pathfinding.rs

```diff
@@ -336,20 +337,23 @@ pub async fn pathfinding_from_train(
     infra: &Infra,
     train_schedule: TrainSchedule,
 ) -> Result<PathfindingResult> {
-    let rolling_stocks =
-        RollingStockModel::retrieve(conn, train_schedule.rolling_stock_name.clone())
-            .await?
-            .into_iter()
-            .map(|rs| (rs.name.clone(), rs))
-            .collect();
+    let rolling_stock = RollingStockModel::retrieve_or_fail(
+        conn,
+        train_schedule.rolling_stock_name.clone(),
+        || PathfindingError::RollingStockNotFound {
+            rolling_stock_name: train_schedule.rolling_stock_name.clone(),
+        },
+    )
+    .await?
+    .into();
```

Comment:
Here it's working because we handle a single train. To avoid confusion and for consistency, I would keep the previous workflow.